framework/src/main/java/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -76,6 +79,26 @@ protected void commonAssignmentCheck(
                     getTypeFactory().createIntRangeAnnotation(Range.CHAR_EVERYTHING));
         }
 
+        if (valueTree.getKind() == Kind.METHOD_INVOCATION
+                && TreeUtils.isArrayscopyOfMethodInvocation((MethodInvocationTree) valueTree)
+                && valueType.getKind() == TypeKind.ARRAY) {
+            List<? extends ExpressionTree> args = ((MethodInvocationTree) valueTree).getArguments();
+            AnnotatedTypeMirror arrType = atypeFactory.getAnnotatedType(args.get(0));
+            if (TreeUtils.isArrayLengthAccess(args.get(1))) {
+                valueType = arrType;
+            } else if (atypeFactory.getAnnotatedType(args.get(1)).getAnnotation(IntVal.class)
+                    != null) {
```

Comment:
Why only `IntVal`? If there is an `IntRange` annotation, you could take its lower bound (`from` field) and use that.

Reply:
(and, you should add a test with `IntRange` when you do)